packages/mattermost-redux/src/types/actions.ts

```diff
@@ -1,30 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {ThunkAction} from 'redux-thunk';
 import {AnyAction} from 'redux';
 import {BatchAction} from 'redux-batched-actions';
 
 import {GlobalState} from '@mattermost/types/store';
+import {ServerError} from '@mattermost/types/errors';
 
 export type GetStateFunc = () => GlobalState;
 export type GenericAction = AnyAction;
 export type Thunk = (b: DispatchFunc, a: GetStateFunc) => Promise<ActionResult> | ActionResult;
 
 export type Action = GenericAction | Thunk | BatchAction | ActionFunc;
 
-export type ActionResult<Data = any, Error = any> = {
+export type ActionResult<Data = unknown, Error = unknown> = {
     data?: Data;
     error?: Error;
 };
 
-export type DispatchFunc = (action: Action, getState?: GetStateFunc | null) => Promise<ActionResult>;
+export type DispatchFunc = (
+    action: Action,
+    getState?: GetStateFunc | null
+) => Promise<ActionResult>;
```

Comment:
I've looked at this myself a few times, and the last version I got to was something like
```ts
export type AppDispatch<State = GlobalState> = ThunkDispatch<State, never, AnyAction>;
export type AppGetState<State = GlobalState> = () => State;
```

I forget why I went with a different name there, but I think relying on `ThunkDispatch` made things easier with handling both regular and thunk actions. Also, by parameterizing `State`, that'll let us redefine a new version of `DispatchFunc` and `GetStateFunc` in the web app or in any other area that extends the state